### PR TITLE
Removed active count and shuffling cache

### PR DESCRIPTION
**Pull request by terencechain**
Following #4264 

This PR removed unused `active count cache` and `shuffling cache`

Change list:
- Removed active indices cache
- Removed active count cache
- Removed all the instances for `helpers.ClearAllCaches()`, it's no longer needed
- Fixed all the tests

### CODE REVIEW

File: shared/testutil/block_test.go [hunk 1]
```diff
@@ -45,7 +44,7 @@ func TestGenerateFullBlock_ThousandValidators(t *testing.T) {
 }
 
 func TestGenerateFullBlock_Passes4Epochs(t *testing.T) {
-	helpers.ClearAllCaches()
+
```
Comment (0xKiwi) [inline shared/testutil/block_test.go]: Are these empty lines at the start of the function intentional? I think we usually don't have an empty line at the start of a function.

Comment (terencechain) [reply shared/testutil/block_test.go]: I agree with you, the new lines show inconsistency. Fixed in the latest commit

File: beacon-chain/core/blocks/block_operations_test.go [hunk 2]
```diff
@@ -1299,7 +1274,6 @@ func TestVerifyIndexedAttestation_OK(t *testing.T) {
 	}
 
 	for _, tt := range tests {
-		helpers.ClearAllCaches()
```
Comment (0xKiwi) [inline beacon-chain/core/blocks/block_operations_test.go]:
```suggestion
```

File: beacon-chain/core/epoch/epoch_processing_test.go [hunk 3]
```diff
@@ -107,7 +107,6 @@ func TestUnslashedAttestingIndices_DuplicatedAttestations(t *testing.T) {
 }
 
 func TestAttestingBalance_CorrectBalance(t *testing.T) {
-	helpers.ClearAllCaches()
```
Comment (0xKiwi) [inline beacon-chain/core/epoch/epoch_processing_test.go]: This one also

File: beacon-chain/core/blocks/block_operations_test.go [hunk 4]
```diff
@@ -1216,7 +1194,6 @@ func TestConvertToIndexed_OK(t *testing.T) {
 		},
 	}
 	for _, tt := range tests {
-		helpers.ClearAllCaches()
```
Comment (0xKiwi) [inline beacon-chain/core/blocks/block_operations_test.go]: Here too

File: beacon-chain/core/epoch/epoch_processing_test.go [hunk 5]
```diff
@@ -315,7 +310,7 @@ func TestBaseReward_AccurateRewards(t *testing.T) {
 		{40 * 1e9, params.BeaconConfig().MaxEffectiveBalance, 2862174},
 	}
 	for _, tt := range tests {
-		helpers.ClearAllCaches()
+
```
Comment (0xKiwi) [inline beacon-chain/core/epoch/epoch_processing_test.go]: Here as well please

File: beacon-chain/core/epoch/precompute/slashing_test.go [hunk 6]
```diff
@@ -97,7 +96,6 @@ func TestProcessSlashingsPrecompute_SlashedLess(t *testing.T) {
 
 	for i, tt := range tests {
 		t.Run(string(i), func(t *testing.T) {
-			helpers.ClearAllCaches()
```
Comment (0xKiwi) [inline beacon-chain/core/epoch/precompute/slashing_test.go]: Here as well

File: beacon-chain/core/state/transition_test.go [hunk 7]
```diff
@@ -100,7 +99,6 @@ func TestExecuteStateTransition_FullProcess(t *testing.T) {
 }
 
 func TestProcessBlock_IncorrectProposerSlashing(t *testing.T) {
-	helpers.ClearAllCaches()
```
Comment (0xKiwi) [inline beacon-chain/core/state/transition_test.go]: Missing a few around here

File: shared/testutil/helpers_test.go [hunk 8]
```diff
@@ -11,7 +11,6 @@ import (
 )
 
 func TestBlockSignature(t *testing.T) {
-	helpers.ClearAllCaches()
```
Comment (0xKiwi) [inline shared/testutil/helpers_test.go]: And here

File: beacon-chain/core/epoch/epoch_processing_test.go [hunk 9]
```diff
@@ -419,7 +411,6 @@ func TestProcessSlashings_SlashedLess(t *testing.T) {
 
 	for i, tt := range tests {
 		t.Run(string(i), func(t *testing.T) {
-			helpers.ClearAllCaches()
```
Comment (0xKiwi) [inline beacon-chain/core/epoch/epoch_processing_test.go]: Last one (?)

File: beacon-chain/core/state/transition_test.go [hunk 10]
```diff
@@ -37,7 +37,6 @@ func TestExecuteStateTransition_IncorrectSlot(t *testing.T) {
 }
 
 func TestExecuteStateTransition_FullProcess(t *testing.T) {
-	helpers.ClearAllCaches()
```
Comment (0xKiwi) [inline beacon-chain/core/state/transition_test.go]: This one

File: beacon-chain/core/state/transition_test.go [hunk 11]
```diff
@@ -181,7 +178,6 @@ func TestProcessBlock_IncorrectProcessBlockAttestations(t *testing.T) {
 }
 
 func TestProcessBlock_IncorrectProcessExits(t *testing.T) {
-	helpers.ClearAllCaches()
```
Comment (0xKiwi) [inline beacon-chain/core/state/transition_test.go]: Over here

**Comment by prestonvanloon** [comment]
Please deprecate the feature flags.

**Pull request merged**